dash/include/dash/HaloMatrix.h

```diff
@@ -0,0 +1,259 @@
+#ifndef DASH__HALOMATRIX_H_INCLUDED
+#define DASH__HALOMATRIX_H_INCLUDED
+
+#include <dash/dart/if/dart.h>
+
+//#include <dash/Team.h>
+#include <dash/Pattern.h>
+//#include <dash/GlobRef.h>
+#include <dash/GlobMem.h>
+//#include <dash/Allocator.h>
+#include <dash/iterator/HaloMatrixIterator.h>
+#include <dash/Halo.h>
+#include <dash/Matrix.h>
+#include <type_traits>
+
+namespace dash {
+
+template<typename MatrixT, typename HaloSpecT>
+class HaloMatrix
```

Comment:
Furthermore it's cumbersome to use it, as no access to the underlying container is provided. Hence the halo has to be created on each iteration (see [here](https://github.com/dash-project/dash/blob/feat-halo-ex-stencil/dash/examples/ex.11.halo-stencil/main.cpp#L120)).

For me, a halo is a prime example for the decorator pattern, because the halo adds features to a container. It's not really an algorithm. However halos should support all containers with block views.